google-cloud-clients/google-cloud-firestore/src/test/java/com/google/cloud/firestore/it/ITSystemTest.java

```diff
@@ -200,6 +200,19 @@ public void serverTimestamp() throws Exception {
     assertTrue(documentSnapshot.getDate("time").getTime() > 0);
   }
 
+  @Test
+  public void updateMicrosecondTimestamp() throws Exception {
+    DocumentReference documentReference = addDocument("time", Timestamp.ofTimeSecondsAndNanos(0, 123000));
+    DocumentSnapshot documentSnapshot = documentReference.get().get();
+
+    Timestamp timestamp = documentSnapshot.getTimestamp("time");
+    documentReference.update("time", timestamp);
+
+    documentSnapshot = documentReference.get().get();
+    timestamp = documentSnapshot.getTimestamp("time");
+    assertEquals(123000, timestamp.getNanos());
```

Comment:
Optional: to test truncation, could you use a timestamp with nanosecond precision and make sure the nanosecond part is truncated when it's added to the database?

Reply:
I really only want to test non-truncation. I hope the server stops truncating altogether at some point, but for now, I can only test the non-truncation using microseconds.